pkg/resources/cloudcontroller/util.go

```diff
@@ -126,6 +126,13 @@ func MigrationToExternalCloudControllerSupported(dc *kubermaticv1.Datacenter, cl
 		}
 		return versionSupported && AzureCloudControllerSupported(v)
 
+	case cluster.Spec.Cloud.Digitalocean != nil:
+		versionSupported, err := version.IsSupported(v.Semver(), kubermaticv1.DigitaloceanCloudProvider, incompatibilities, kubermaticv1.ExternalCloudProviderCondition)
+		if err != nil {
+			return false
+		}
+		return versionSupported && DigitalOceanCloudControllerSupported(v)
```

Comment:
This is not needed because we don't have CCM/CSI migration for DO. Instead, you need to add DigitalOcean to the function above (`ExternalCloudControllerFeatureSupported`).